### ISSUE TITLE
multiple RNA libraries

### ISSUE THREAD

**Issue by michieitel** [ksahlin/BESST_RNA#8]
Hi. I have several paired-end RNA libraries with different library statistics (insert size, read length [also various length each after trimming]). How can I input these in the tool? Does besst_rna calculate the values (-m, -s, -r) for each or does it generate average values if I use the default settings?? How can I enter multiple libraries?

I estimated the library stats for the bam file of all libraries mapped at once with hisat2:
SN	average length:	131
SN	insert size average:	222.1
SN	insert size standard deviation:	250.1
SN	pairs on different chromosomes:	143564

Should/Can I specify '-r 131 -s 250 -m 222' ?? Or is it better to scaffold with each library individually one after another?

Thanks
Michael

**Comment by michieitel** [comment]
one addition: I estimated the library stats for the bam file of all libraries mapped at once with hisat2 ...
SN	average length:	131
SN	insert size average:	222.1
SN	insert size standard deviation:	250.1
SN	pairs on different chromosomes:	143564

can I specify  '-r 131 -s 250 -m 222' ??

**Comment by ksahlin** [comment]
How much does the mean insert sizes vary roughly? 

BESST_RNA can only take one library. So, assuming your insert sized doesn't vary in the order of thousands,  it's best if you merge the bam files of all libraries and submit it as one bam file all together. BESST_RNA does not have -m and -s as parameters, because paired reads can span junctions.

Hope this helps,
K

**Comment by michieitel** [comment]
"BESST_RNA does not have -m and -s as parameters"?? Running 'python Main.py 1 ' gives me:
  -s STDDEV, --stddev=STDDEV
                        estimated standard deviation of libraries
  -r READLEN            Mean read length
  -m MEAN               mean of insert library

Anyhow. I mapped all libraries individually and except for one the stats are similar:
library	insert	SD	read_lenth
lib1	266.4	319.8	142
lib2	266.6	320.2	142
lib3	265.9	317.0	142
lib4	265.9	315.0	142
lib5	281.4	411.7	142
lib6	279.8	405.1	142
lib7	281.3	412.3	142
lib8	280.7	408.7	142
lib9	267.0	338.5	143
lib10	266.5	334.9	143
lib11	266.1	333.8	143
lib12	266.3	334.9	143
lib13	269.2	332.8	144
lib14	269.8	338.2	144
lib15	269.1	333.0	144
lib16	269.0	334.4	144
lib17	261.2	314.4	144
lib18	261.0	314.3	144
lib19	260.6	314.3	144
lib20	260.6	311.7	144
lib21	285.3	421.3	144
lib22	287.1	433.0	144
lib23	286.5	429.9	144
lib24	285.7	422.6	144
lib25	189.1	113.3	100
lib26	290.5	987.7	101
lib27	203.1	210.5	100

Therefore I should be able to use all together, right (probably with the exception of lib26)? Does it matter that 3 libs have different read length?

Thanks again!
Michael

**Comment by ksahlin** [comment]
Yes, they should definitely be merged for RNA scaffolding in that case. From what I remember, It should not matter that they have different read lengths.

Oh, this is poor documentation from me. I believe these parameters are not used internally.   BESST_RNA was a branch from BESST (which has `-m` and `-s`) used for a pilot project with (at the time) no intent to publish the software. Due to this, this repo has so far not been very well maintained by me. Apologies for this.

**Comment by michieitel** [comment]
Thanks for your input Kristoffer!

I ran the tool with all libs mapped and it produced these stats:
PASS 1
Nr of contigs that was singeled out due to length constraints 0
Time elapsed for creating graph, iteration 0: 3373.15909314
460 link edges created.
157 isolated contigs removed from graph.
122 ambiguous regions in graph ( a contig with more than 2 neighbors).
50 of these regions solved (one dominating edge)
72 of these regions unsolved (removed all edges)
150 isolated contigs removed from graph.
Contigs/scaffolds left: 121
Remove remaining cycles...
A cycle in the scaffold graph: [(185, 'L'), (303, 'L'), (303, 'R'), (185, 'R')]
A cycle in the scaffold graph: [(185, 'L'), (303, 'L'), (303, 'R'), (185, 'R')]
1 cycles removed from graph.
Nr of new scaffolds created in this step: 53
Time elapsed for making scaffolds, iteration 0: 0.0176231861115

Contigs before scaffolding: 428
(super)Contigs after scaffolding: 362

Do I have to worry about these 122 ambiguous regions?
Michael

**Comment by ksahlin** [comment]
These could be either because of misaligned reads (e.g., from repeats) or possibly because of alternative splicing. 

Whenever there is enough support to select one of the edges and remove the other, BESST_RNA will do so and connect the two contigs with the dominating edge. If they have relatively even support, BESST_RNA will remove all the edges and no connection is formed. 

I can't really answer if you have to worry or not. Looking at the number of links in these region might help to answer that but there is currently no such information in the output. Maybe you have some other way that you to can validate the scaffolds produced?
